bindings/rust/src/lib.rs

```diff
@@ -1135,37 +1045,49 @@ macro_rules! sig_variant_impl {
                                 &[],
                             ) != BLST_ERROR::BLST_SUCCESS
                             {
-                                valid.store(false, Ordering::Relaxed);
+                                valid.store(false, Ordering::Release);
                                 break;
                             }
+
+                            processed += 1;
                         }
-                        if valid.load(Ordering::Relaxed) {
+
+                        if processed > 0 && valid.load(Ordering::Acquire) {
```

Comment:
ditto: relaxed ordering is OK there and will be faster on weak memory models like ARM